Joesdev/dog-messenger
# FASP-46 Adds validation to results page

## Whats This PR Do?
Adds fuctionality which requires a user to have a token in order to access their results page by:
1. Assigning a unique token upon user creation
2. Atatching the same token to a user notification
3. Creating createUserToken and getUserToken functions

Refactors showCollectedArrayOfDogsView by:

1. Removing functionality into new function appendFoundDogCollectionDataToApiData

Adds tests for all new functions listed above

## Steps to Review
1. Generate an email notification
2. Click "show me" button on the above notification
3. Verify the results page is properly rendered
4. Navigate to /results/yourValidEmailHere and verify the page in
